stable/graphite/templates/statefulset.yaml

```diff
@@ -8,25 +8,19 @@ metadata:
     release: {{ .Release.Name }}
     heritage: {{ .Release.Service }}
 spec:
-  strategy:
-    type: Recreate
+  updateStrategy:
+    type: RollingUpdate
```

Comment:
This should be configurable.

Reply:
Why? The only other option will create surprises. If a user performs `helm upgrade` they expect the new version to be deployed. This makes the new behavior match the old, modulo some downtime. (The previous case was also bad - multiple versions running simultaneously!)

Reply:
Recreate isn't an option for stateful sets.

Reply:
There are two options: OnDelete and RollingUpdate. See [StatefulSets - Kubernetes](https://kubernetes.io/docs/concepts/workloads/controllers/statefulset/#update-strategies).

I suspect the problem you had with rolling upgrades is due to persistence. If persistence is enabled, and the volume is ReadWriteOnce, you end up with contention on the disk. That's because a Deployment will happily run more pods than you specify. So the 2nd Pod comes up, can't acquire the persistent volume, and fails to start.

Reply:
I wonder why my colleagues, who use the chart, did'nt complained earlier about this...

Reply:
ReadWriteOnce access mode volumes can still be accessed by multiple pods *on the same node*, so if they were testing on a single node cluster, had affinity configured, or just plain got lucky, they might not have noticed it. ReadWriteOnce is more like "ReadWrite for all pods colocated on the same node". But that's kind of long, so I think they shortened it.